Core/Core.Platform.iOS/Storage/KeychainWrapper.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using Foundation;
+using Security;
+
+namespace AeroGear.Mobile.Core.Storage
+{
+    public class KeychainWrapper
+    {
+        private readonly string KeychainService;
+        private static readonly bool DefaultSynchronize = false;
+
+        public KeychainWrapper(string keychainService)
+        {
+            KeychainService = keychainService;
+        }
+
+        /// <summary>
+        /// Save the specified value using the specified key as an identifier.
+        /// This key can later be used to retrieve and remove the value.
+        /// </summary>
+        /// <returns><c>true</c> if the save operation succeeded. Otherwise <c>false</c>.</returns>
+        /// <param name="key">Identifier.</param>
+        /// <param name="value">Value to store.</param>
+        public bool Save(string key, string value)
+        {
+            return SetPassword(KeychainService, key, value, SecAccessible.Always, DefaultSynchronize) == SecStatusCode.Success;
+        }
+
+        /// <summary>
+        /// Load the value associated with the specified key.
+        /// </summary>
+        /// <returns>The string value associated with the specified key.</returns>
+        /// <param name="key">Key that identifies the stored value.</param>
+        public string Load(string key)
+        {
+            return GetPassword(KeychainService, key, DefaultSynchronize);
+        }
+
+        /// <summary>
+        /// Remove the value associated with the specified key.
+        /// </summary>
+        /// <returns><c>true</c> if the remove operation succeeded. Otherwise <c>false</c>.</returns>
+        /// <param name="key">Identifier.</param>
+        public bool Remove(string key)
+        {
+            return DeletePassword(KeychainService, key, DefaultSynchronize) == SecStatusCode.Success;
+        }
+
+        /// <summary>
+        /// Retrieves a password from the Keychain.
+        /// </summary>
+        /// <returns>The stored password if it exists. Else <c>null</c>.</returns>
+        /// <param name="service">Service.</param>
+        /// <param name="username">Username associated with the password.</param>
+        /// <param name="synchronizable">Whether the item is synchronizable through iCloud.</param>
+        private string GetPassword(string service, string username, bool synchronizable)
+        {
+            if (service == null)
+            {
+                throw new ArgumentNullException("service", "Service cannot be null");
+            }
+            if (username == null)
+            {
+                throw new ArgumentNullException("username", "Username cannot be null");
+            }
```

Comment:
can we use the methods in the SanityCheck class instead?

Reply:
Ah yep, forgot about those